File: Assets/Editor/SceneAPI/Modules/Objects/ResetObjectModule.cs
using System;
using System.Net;
using Newtonsoft.Json;
using UnityEngine;

namespace SceneAPI.Modules
{
    public static class ResetObjectModule
    {
        public static string Execute(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);

                string objectPath = data?.path;
                if (string.IsNullOrEmpty(objectPath))
                {
                    return JsonConvert.SerializeObject(new { success = false, error = "Object path is required" });
                }

                GameObject obj = GameObjectUtilities.FindGameObjectByPath(objectPath);
                if (obj == null)
                {
                    return JsonConvert.SerializeObject(new { success = false, error = "Object not found" });
                }

                // Сбрасываем базовые параметры Transform в дефолт
                if (obj.transform != null)
                {
                    obj.transform.localPosition = Vector3.zero;
                    obj.transform.localRotation = Quaternion.identity;
                    obj.transform.localScale = Vector3.one;
                }

                // Можно расширить: сбрасывать свойства стандартных компонентов (Renderer, Collider, etc.)
                return JsonConvert.SerializeObject(new { success = true, message = "Object reset to defaults" });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new { success = false, error = $"Error resetting object: {ex.Message}" });
            }
        }

        private static string GetRequestBody(HttpListenerContext context)
        {
            using (var reader = new System.IO.StreamReader(context.Request.InputStream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}

File: Assets/PythonMCP/modules/reset_object_module.py
import requests
import json
from typing import Dict

class ResetObjectModule:
    def __init__(self, base_url: str):
        self.base_url = base_url
    
    def execute(self, object_path: str) -> Dict:
        """Сбрасывает параметры объекта на сцене (или компонента/префаба по политике сервера)"""
        try:
            if not object_path:
                return {
                    "success": False,
                    "action": "reset_object",
                    "error": "object_path is required"
                }
            
            data = { "path": object_path }
            response = requests.put(
                f"{self.base_url}/objects/reset",
                json=data
            )
            response.raise_for_status()
            result = response.json()
            return {
                "success": result.get("success", False),
                "action": "reset_object",
                "data": result,
                "error": result.get("error")
            }
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "reset_object",
                "error": f"Request error: {str(e)}"
            }
        except json.JSONDecodeError as e:
            return {
                "success": False,
                "action": "reset_object",
                "error": f"JSON decode error: {str(e)}"
            }